test/api/handlers_test.go

```diff
@@ -302,6 +304,59 @@ func testUserPUT(t *testing.T, at *test.AccountsTester) {
 	}
 }
 
+// testUserDELETE tests the DELETE /user endpoint.
+func testUserDELETE(t *testing.T, at *test.AccountsTester) {
+	u, c, err := test.CreateUserAndLogin(at, t.Name())
+	if err != nil {
+		t.Fatal("Failed to create a user and log in:", err)
+	}
+
```

Comment:
Building from my other comment, we should add a test case here to delete the user before adding test data.